### PR TITLE
Delete endpoint

### PR DESCRIPTION
**Pull request by matas-cc**
### Description
It has been decided to use hard deletes for access objects.

If versioning is enabled on a S3 bucket, the object itself will not be deleted but a delete marker will be placed.
https://docs.aws.amazon.com/AmazonS3/latest/userguide/DeletingObjectVersions.html

### Parent Issue
https://github.com/cyber-crypt-com/encryptonize-core/issues/187

### Type(s) of Change (Split the PR if you check many)
- [x] Added user feature
- [x] Added technical feature
- [x] Added tests
- [ ] Refactor
- [ ] Bugfix
- [x] Updated documentation
- [ ] Infrastructure changes (Terraform, GCP, K8s, other)
- [ ] CI/CD workflow changes

### Testing Checklist
- [ ] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [ ] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make docker-up; make tests`.
- [ ] I have updated relevant workflows and deployment tools.
- [x] I have updated/added relevant documentation (readme, doc comments, etc).
- [ ] I have added the license to new source code files.
- [x] I have spent some time looking over the full diff before creating this PR.

### Technical Debt
Concurrency issues, similar to https://github.com/cyber-crypt-com/encryptonize-core/pull/189

### CODE REVIEW

File: encryption-service/tests/grpce2e/delete_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	defer closeClient(client, t)
+
+	plaintext := []byte("foo")
+	associatedData := []byte("bar")
+
+	storeResponse, err := client.Store(plaintext, associatedData)
+	failOnError("Store operation failed", err, t)
+	oid := storeResponse.ObjectId
+	_, err = client.Delete(oid)
+	failOnError("Delete operation failed", err, t)
```
Comment (philip-cc) [inline encryption-service/tests/grpce2e/delete_test.go:33]: We should probably try to retrieve the object here to check if it was actually deleted.

File: documentation/api/api-v2.md [hunk 2]
```diff
@@ -211,6 +221,15 @@ the Encryption Service cannot reach the object storage. In these cases, an error
...
 
+# Delete
+
+Deletes a previously Stored `Object`. This call does not fail if the specified object does not exist. It can fail if the caller does not have access permission to that object or if
+the Encryption Service cannot reach the object storage. In these cases, an error is returned.
+
+```
+rpc Delete (DeleteRequest) returns (DeleteResponse)
+```
+
```
[the remark this answers is not in the document]
Comment (matas-cc) [reply documentation/api/api-v2.md:232]: True, fixed

File: encryption-service/services/enc/storage_handlers.go [hunk 3]
```diff
@@ -127,3 +128,37 @@ func (enc *Enc) Retrieve(ctx context.Context, request *RetrieveRequest) (*Retrie
...
+	_, err := AuthorizeWrapper(ctx, enc.Authorizer, objectIDString)
+	if err != nil {
+		// AuthorizeWrapper logs and generates user facing error, just pass it on here
+		return nil, err
+	}
+
+	// Parse objectID from request
+	objectID, err := uuid.FromString(objectIDString)
+	if err != nil {
+		errMsg := fmt.Sprintf("AuthorizeWrapper: Failed to parse object ID %s as UUID", objectIDString)
```
[the remark this answers is not in the document]
Comment (matas-cc) [reply encryption-service/services/enc/storage_handlers.go:146]: fixed

[the remark this answers is not in the document]
Comment (matas-cc) [reply encryption-service/services/enc/storage_handlers.go:146]: :facepalm:  fixed

File: encryption-service/impl/objectstorage/objectstorage.go [hunk 4]
```diff
@@ -19,10 +19,12 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"errors"
-	"github.com/gofrs/uuid"
 	"net/http"
 
+	"github.com/gofrs/uuid"
+
 	"encryption-service/contextkeys"
```
[the remark this answers is not in the document]
Comment (matas-cc) [reply encryption-service/impl/objectstorage/objectstorage.go:24]: fixed

**Review by philip-cc — APPROVED** [review]

**Pull request merged**